lib/flake-utils.nix

```diff
@@ -0,0 +1,63 @@
+{ nixpkgs, systems }:
+rec {
```

Comment:
I have been told using `rec` is an anti-pattern (I guess because it can fail in obscure ways) and it's usage is discouraged.

You are a heavy nix contributor and seem to prefer it. Ergo disagree?

Reply:
It's mainly the nix.dev guide discouraging it. It's used in most nixpkgs and in all library files in ./lib. I don't consider it a problem.